packages/lib-auth/test/functional/node/isolation/test.js

```diff
@@ -0,0 +1,118 @@
+const nock = require('nock')
+
+const createClient = require('../../../../src')
+const fixtures = require('./fixtures')
+
+describe('Functional Tests > Node (Isolation)', function () {
+  let client
+  let scope
+
+  before(function () {
+    client = createClient({
+      clientAppID: fixtures.clientAppID,
+      hostUrl: fixtures.baseUrl
+    })
+  })
+
+  describe('user registration', function () {
+    let scope
+
+    before(function () {
+      scope = nock(fixtures.baseUrl)
+        .persist()
+        .head('/users/sign_in')
+        .query(queryObject => !!queryObject.now)
+        .reply(200, '', {
+          'x-csrf-token': fixtures.mockCSRFToken
+        })
+        .post('/users')
+        .reply(200)
+        .post('/users/sign_in')
+        .reply(200, '', {
+          'set-cookie': [
+            `_Panoptes_session=${fixtures.mockJWT}`
+          ]
+        })
+        .post('/oauth/token')
+        .reply(200, {
+          'access_token': fixtures.mockAccessToken,
+          'created_at': fixtures.mockCreatedAt,
+          'expires_in': fixtures.mockAccessTokenExpiresIn,
+          'refresh_token': fixtures.mockRefreshToken,
+          scope: fixtures.mockScope
+        })
+    })
+
+    after(function () {
+      nock.cleanAll()
+    })
+
+    it('should let you register as a new user', async function () {
+      const newToken = await client.register(fixtures.newUser)
+      expect(newToken).to.equal(fixtures.mockAccessToken)
+    })
```

Comment:
We should test a failure case too. What is the expected behavior if the server returns a 400 or 500 range response?

Reply:
Anyone know,  off the top of their head, what the response is when registration fails (user already exists)? We should have a test for that response.

Reply:
Looks like 422 if the built user resource can't be saved, https://github.com/zooniverse/Panoptes/blob/99857e1554608434c7696a26ea725d09f9b06474/app/controllers/registrations_controller.rb#L75

It might 500 if a db uniq index has a duplicate value on save.

Reply:
Cheers! I'll add tests for both.